tools/upgrade-to-1.1.0.sh

```diff
@@ -0,0 +1,212 @@
+        ;;
+      manifest_file)
+        eval "${param}"="${val}"
+        ;;
+      help)
+        help && exit 0
+        ;;
+      *)
+        echo "Parameter not found: '$param'"
+        help && exit 1
+        ;;
+    esac
+  done
+
+  if [[ -z "$current_namespace" ]]; then
+    echo "Must provide current_namespace" 1>&2
+    help
+    exit 1
+  fi
+  if [[ -z "$manifest_file" ]]; then
+    echo "Must provide manifest_file" 1>&2
+    help
+    exit 1
+  fi
+ 
+}
+
+namespace_exist() {
+    echo "check if namespace ${current_namespace} exist"
+    kubectl get namespace ${current_namespace}
+    if [ $? != 0 ]
+    then
+        echo "namespace ${current_namespace} do not exist, exiting..."
+        exit 13 
+    fi
+}
+
+backup() {
+    backup_file_name=backup_${CURRENT_TIME}.yaml
+    echo "--------------------------------------------------------------"
+    echo "taking backup of existing operator installation manifest files"
+    echo "backup namespace"
+    echo "---" >> ${backup_file_name}
+    kubectl get namespace ${current_namespace} -o yaml >> ${backup_file_name}
+    if [ $? == 0 ] 
+    then 
+        echo "" >> ${backup_file_name}
+        echo "---" >> ${backup_file_name}
+    fi
+    echo "backup serviceaccount details"
+    kubectl get serviceaccount ${current_namespace} -n splunk-operator -o yaml >> ${backup_file_name}
+    if [ $? == 0 ] 
+    then 
+        echo "" >> ${backup_file_name}
+        echo "---" >> ${backup_file_name}
+    fi
+    echo "backup if there are any role defined for splunk operator"
+    kubectl get role splunk:operator:namespace-manager -n ${current_namespace} -o yaml >> ${backup_file_name}
+    if [ $? == 0 ] 
+    then 
+        echo "" >> ${backup_file_name}
+        echo "---" >> ${backup_file_name}
+    fi
+    echo "backup if there are any role-biding defined for splunk operator"
+    kubectl get rolebinding splunk:operator:namespace-manager -n ${current_namespace}  -o yaml >> ${backup_file_name}
+    if [ $? == 0 ] 
+    then 
+        echo "" >> ${backup_file_name}
+        echo "---" >> ${backup_file_name}
+    fi
+    echo "backup if there are any cluster-role defined for splunk operator"
+    kubectl get clusterrole splunk:operator:resource-manager -o yaml  >> ${backup_file_name}
+    if [ $? == 0 ] 
+    then 
+        echo "" >> ${backup_file_name}
+        echo "---" >> ${backup_file_name}
+    fi
+    echo "backup if there are any cluster-role-binding defined for splunk operator"
+    kubectl get clusterrolebinding splunk:operator:resource-manager -o yaml >> ${backup_file_name}
+    if [ $? == 0 ] 
+    then 
+        echo "" >> ${backup_file_name}
+        echo "---" >> ${backup_file_name}
+    fi
+    echo "backup deployment details"
+    kubectl get deployment splunk-operator -n ${current_namespace} -o yaml >> ${backup_file_name}
+    if [ $? == 0 ] 
+    then 
+        echo "" >> ${backup_file_name}
+        echo "---" >> ${backup_file_name}
+    fi
```

Comment:
Don't we need to fail and exit(probably cleanup?), in case if any of these commands fail?

Reply:
no its just backup, both role and cluster role will not there together. so one of them will fail here.